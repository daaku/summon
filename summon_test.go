package summon_test

import (
	"context"
	"testing"

	"github.com/daaku/ensure"
	"github.com/daaku/summon"
	"github.com/gkampitakis/go-snaps/snaps"
)

func TestCheckInternet(t *testing.T) {
	t.Parallel()
	ensure.Nil(t, summon.Run(context.Background(), summon.CheckInternet))
}

func TestShellf(t *testing.T) {
	cases := []struct {
		name   string
		format string
		args   []any
	}{
		{name: "lone command", format: "true"},
		{name: "fixed args", format: "echo hello world"},
		{name: "arg with space", format: "echo %q", args: []any{"hello world"}},
		{name: "path quoting", format: "echo %q/foo/bar", args: []any{"/home/naitik shah"}},
	}
	for _, c := range cases {
		t.Run(c.name, func(t *testing.T) {
			name, args, err := summon.Shellf(c.format, c.args...)
			snaps.MatchSnapshot(t, name, args, err)
		})
	}
}
