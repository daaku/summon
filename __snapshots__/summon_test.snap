
[TestShellf/lone_command - 1]
true
[]string{}
nil
---

[TestShellf/fixed_args - 1]
echo
[]string{"hello", "world"}
nil
---

[TestShellf/arg_with_space - 1]
echo
[]string{"hello world"}
nil
---

[TestShellf/path_quoting - 1]
echo
[]string{"/home/naitik shah/foo/bar"}
nil
---
